### DISCUSSION TITLE
Is Windows storing the last used Connection Parameters for a device?

### DISCUSSION THREAD

**Discussion by ChrisM0815** [hbldh/bleak#1215] [Q&A] [answered]
I'm using a BLE-Device that changes the Connection Parameters (CP) during operation. If I restart the bleak script, the same connection parameters are used that were requested by the device when I ran the script beforehand. If I restart my pc or use a different BLE device, the default CP are used again when I run the script. I didn't observer this behavior on Linux.  Does Windows store the previously used CP for a specific device?

**Comment by dlech** [comment, marked as answer]
The Windows Bluetooth stack is a mystery to most of us. You would need to ask one of the Microsoft engineers that actually worked on it to find out.
